Pd 66 scale write new virtualization apps configuration article (#1717)
truenas/documentation: content/SCALE/GettingStarted/VMandAppConfigSCALE.md

---

> title: "Configuring Virtualization and Apps in TrueNAS SCALE"

Reviewer: The description: "" parameter is missing

Author: It has been added.

weight: 50
tags:
- scalevm
- scaleapps
- scalenetwork
- scalevlan
- scalelagg
- scaledatashare
---

{{< toc >}}

> Configuring TrueNAS SCALE to work with virtualized features, such as VMs and Apps, is a part of the setup process that when optimized takes advantage of the network storage capabilities that SCALE offers. 

Reviewer: ```suggestion
Configuring TrueNAS SCALE to work with virtualized features, such as virtual machines (VMs) and applications, is a part of the setup process that when optimized takes advantage of the network storage capabilities that SCALE offers. 
```

## Before You Begin

> This article assumes that you have the latest release version of TrueNAS SCALE installed on your system. The following steps comprise a list of configuration prerequisites that you have completed and are familiar with before beginning VM and App installation.

Reviewer: if you are talking about the UI option Apps then it goes in bold, otherwise not in caps.
```suggestion
This article assumes that you have the latest release version of TrueNAS SCALE installed on your system. The following steps comprise a list of configuration prerequisites that you have completed and are familiar with before beginning VM and app installations.
```

### Network Prerequisites

The primary network interface is configured as part of the SCALE installation process. Go to **Network > Global Configuration** screen in the TrueNAS web UI to determine that the default gateway, host name, domain and DNS name servers have been configured correctly. See [Console Setup Menu Configuration]({{< relref "ConsoleSetupMenuScale.md" >}}) for more information on network settings.

> ### Optional: Network VLAN 

Reviewer: This is not parallel construction for the headers. You can consider changing all header to something like this: 
### Network Configuration
### VLAN Configuration
### Storage Configuration
### Shares Configuration

Author: Headers have been changed.

You have the option of configuring a virtual LAN (VLAN) to route traffic for your VMs. VLAN benefits include the reduction of broadcast traffic and the ability to group resources in different physical locations into a broadcast domain. VLANs virtually segment a network. Different VLANs can communicate with each other using layer 3 devices. See [Setting Up a Network VLAN]({{< relref "SettingUpVLAN.md" >}}) for more information on creating virtual LANs (VLAN).

### Storage Prerequisites

Storage pool creation is part of the initial process of setting up storage for SCALE. A TrueNAS dataset is a file system within a data storage pool. See [Setting Up Storage]({{< relref "SetUpStorageScale.md" >}}) to review storage pool creation and [Adding and Managing Datasets]({{< relref "DatasetsSCALE.md" >}}) for information on dataset configuration. 

### Set Up Shares

> Once the pool and datasets have been created, set up shares to enable data access. Three types of data sharing methods are discussed in 

Reviewer: Rather than using once, use after because the action occurs after you do something.
```suggestion
After creating the pool and datasets, set up shares to enable data access. Three types of data sharing methods are discussed in 
```

> [Setting Up Data Sharing]({{< relref "SetUpSharing.md" >}}). More specific coverage of each share should be investigated based on your individual use case. 

Reviewer: Why the carriage return for this relref? Also
```suggestion
[Setting Up Data Sharing]({{< relref "SetUpSharing.md" >}}). You should investigate more specific coverage of each share based on your individual use case. 
```

> A good introduction as to how TrueNAS SCALE handles SMB shares is located at [SMB Shares Screens]({{< relref "SMBSharesScreens.md" >}}) and [Setting Up SMB Home Shares]({{< relref "AddSMBHomeShare.md" >}}). For a better understanding of how to configure Active Directory, see [Active Directory Screen]({{< relref "scale/scaleuireference/credentials/directoryservices/ActiveDirectory.md" >}}). Required Kerberos information is outlined here, and also in this article: [Configuring Kerberos]({{< relref "ConfigKerberosSCALE.md" >}}). Configuring LDAP best practices can be found at: [Configuring LDAP]({{< relref "ConfigLDAPSCALE.md" >}}).

Reviewer: Better organization of this section, maybe try
```suggestion
[SMB Shares Screens]({{< relref "SMBSharesScreens.md" >}}) and [Setting Up SMB Home Shares]({{< relref "AddSMBHomeShare.md" >}}) provide a good introduction as to how TrueNAS SCALE handles SMB shares. See [Active Directory Screen]({{< relref "scale/scaleuireference/credentials/directoryservices/ActiveDirectory.md" >}}) for a better understanding of how to configure Active Directory, and [Configuring Kerberos]({{< relref "ConfigKerberosSCALE.md" >}}) for an outline of required Kerberos information. For LDAP best practices see [Configuring LDAP]({{< relref "ConfigLDAPSCALE.md" >}}).
```
Also, it is a bit of a non-sequitur to lump shares and directory services together in this paragraph and before you mention NFS shares. They are not the same thing so this is a bit confusing. Make the directory services information a separate paragraph or a new heading such as 
### Directory Services Configuration

Author: Shares content has been updated. Directory services are now under a separate header.

> NFS basic share creation information can be found here: [Adding NFS Shares]({{< relref "AddingNFSShares.md" >}}). Adjust access permissions using the advanced options. 

Reviewer: ```suggestion
See [Adding NFS Shares]({{< relref "AddingNFSShares.md" >}}) for information on creating a basic NFS share. Adjust access permissions using the advanced options. 
```

Author: Content has been updated.

## Virtualization Prerequisites

> In order to be able to run a Virtual Machine (VM), hardware requirements include an x86 machine running a recent Linux kernel using an Intel processor with VT extensions, or an AMD processor with SVM extensions (AMD-V). To install a VM on SCALE, you need to first research the minimum and recommended specifications for the OS you plan to use and your full use case for that VM. Allocating too many resources to a VM can cause performance on the TrueNAS SCALE system to suffer, so it is recommended to plan ahead and ensure that the SCALE system has the resources to run both itself and a full VM deployment effectively.

Reviewer: A few suggestions to make this easier to understand
```suggestion
To run a virtual machine (VM), hardware requirements include an x86 machine running a recent Linux kernel using either an Intel processor with VT extensions or an AMD processor with SVM extensions (AMD-V). 
To install a SCALE VM, first research the minimum and recommended specifications for the OS you plan to use and for your full use case for that VM. 
Allocating too many resources to a SCALE VM can cause performance to suffer, so we recommend you plan ahead and ensure your SCALE system has the resources to run both itself and a full VM deployment effectively.
```

> Software requirements include an installer for the OS you intend to install on the VM, and a TrueNAS storage pool. It is recommended to have additional datasets created beneath the storage pool to further organize your VM data.

Reviewer: Why are software and storage lumped together in this paragraph. What is the connection?
Suggest separating to separate paragraphs or include what the connection between the two are.

Author: I have separated software and storage requirements as suggested.

> Review the [Virtualization Screens]({{< relref "VirtualizationScreens.md" >}}) to determine requirements for VM installation. See [Adding Managing VMs]({{< relref "CreatingManagingVMsSCALE.md" >}}) for more information on adding or managing VMs.

Reviewer: ```suggestion
Review [Virtualization Screens]({{< relref "VirtualizationScreens.md" >}}) to determine requirements for VM installation. See [Adding Managing VMs]({{< relref "CreatingManagingVMsSCALE.md" >}}) for more information on adding or managing VMs.
```

> ## App Installation Prerequisites

Reviewer: ```suggestion
## Application Installation Prerequisites
```

> The first time you open the **Applications** screen, the UI asks you to choose a storage pool for applications. Select a pool that has enough space for all the application containers you intend to use. Set up a new dataset before installing your applications if you want to store your application data in a location separated from other storage on your system.

Reviewer: Let the user know they don't have to set the pool the first time they access the Applications screen. They might not know they can cancel out of this process.
Also, you might want to let them know they can create their datasets before they go to this screen if that is what they want to do.
```suggestion
The first time you open the **Applications** screen, the UI asks you to choose a storage pool for applications. Either select a pool with enough space for all the applications you intend to use or you can cancel out of this and select the pool when you have all your storage set up the way you want it. Set up a new dataset before installing your applications if you want to store your application data in a location separated from other storage on your system.
```

Author: Revised to add additional information about pool selection.

![AppsSettingsChoosePool](/images/SCALE/22.02/AppsSettingsChoosePool.png "Choosing a Pool for Apps")

**Applications** > **Available Applications** displays official applications pre-configured for TrueNAS SCALE. These have fewer requirements for deployment than custom applications. See [Using Apps]({{< relref "UsingApps.md" >}}) for more information, as well as [Application Screens]({{< relref "AppsScreensSCALE.md" >}}).

For custom applications, [Launch Docker Image Screens]({{< relref "LaunchDockerImageScreens.md" >}}) details each field in the **Launch Docker Image** wizard. Prior to beginning a custom App installation, determine the following information:

> * Your host networking requirements for the container including host interface

Reviewer: Are you wanting to create a list of requirements here? What is the fist list item? host interface? Also, watch for parallel construction in this list.
This section is not clear

Author: Revised list of requirements to reflect parallel construction, and make content meaning clearer.
